content/data-loading.md

```diff
@@ -0,0 +1,556 @@
+
+It's also worth knowing a little about what happens on the server when the new subscription is started and the old one is stopped.
+
+The server *explicitly* waits until all the data is sent down (the new subscription is ready) for the new subscription before removing the data from the old subscription. The idea here is to avoid flicker -- you can, if desired, continue to show the old subscription's data until the new data is ready, then instantly switch over to the new subscription's complete data set.
+
+What this means is in general, when changing subscriptions, there'll be a period where you are *over-subscribed* and there is more data on the client than you strictly asked for. This is one very important reason why you should always fetch the same data that you have subscribed to (don't "over-fetch").
+
+### Paginating subscriptions
+
+A very common pattern of data access is pagination. This refers to the practice of fetching a ordered list of data one "page" at a time --- typically some number of items, say twenty.
+
+There are two styles of pagination that are commonly used, a "page-by-page" style---where you show only one page of results at a time, starting at some offset (which the user can control), and "infinite-scroll" style, where you show and increasing number of pages of items, as the user moves through the list (this is the typical "feed" style user interface).
+
+Let's consider a publication/subscription technique for the second (the first technique is quite difficult to do correctly in Meteor as you can't tell at which offset to start on the client-side which only has a subset of the true data).
+
+XXX: Should we deal with the page-to-page pagination scenario better?
+
+In an infinite scroll publication, we simply need to add a new argument to our publication controlling how many items to load. Suppose we wanted to paginate the todo items in our Todos example app:
+
+```js
+const MAX_TODOS = 1000;
+
+Meteor.publish('list/todos', function(listId, limit) {
+  check(listId, String);
+  check(limit, Number);
+
+  const options = {
+    sort: {createdAt: -1},
+    limit: Math.min(limit, MAX_TODOS)
+  };
+
+  ...
+});
+```
+
+It's important that we set a `sort` parameter on our query (after, which first `limit` todos do we want?), and that we set an absolute maximum on the number of items a user can request (at least in the case where lists can grow without bound).
+
+Then on the client side, we'd some kind of reactive state variable to control how many items to request:
+
+```js
+Template.listsShowPage.onCreated(function() {
+  this.state = new ReactiveDict();
+  this.autorun(() => {
+    this.state.set('listId', FlowRouter.getParam('_id'));
+    this.subscribe('list/todos', this.state.get('listId'), this.state.get('requestedTodos'));
+  });
+});
+```
+
+We'd increment that `requestedTodos` variable when the user clicks "load more" (or perhaps just when they scroll to the bottom of the page).
```

Comment:
Isn't this potentially really inefficient? Like as far as I know, the client will get rid of all of the old documents and send all of them over again? Or does the merge box handle it somehow?